ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/explorer/project/SearchNodeHandler.java

```diff
@@ -58,8 +59,18 @@ public SearchNodeHandler(Tree tree) {
         tree.getNodeLoader().addPostLoadHandler(this);
     }
 
-    public Promise<Node> getNodeByPath(final HasStorablePath path, boolean forceUpdate) {
+    /**
+     * Search node in the project explorer tree by storable path.
+     *
+     * @param path path to node
+     * @param forceUpdate force children reload
+     * @param closeMissingFiles allow editor to close removed files if they were opened
+     * @return
+     * promise object with found node or promise error if node wasn't found
+     */
+    public Promise<Node> getNodeByPath(final HasStorablePath path, boolean forceUpdate, boolean closeMissingFiles) {
```

Comment:
closeMissingFiles parameter holds simple flag to pass it into project explorer module after node found. Business logic on file closing need to be moved from project explorer module into search node handler, but it may took few hours. Thats why as temporary solutions I decided to do it in this way